### PR TITLE
FIX: Make the keyboard shortcuts compatible with modal using the new API

### PR DESCRIPTION
**Pull request by megothss**

### CODE REVIEW

File: assets/javascripts/discourse/services/d-templates-modal.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  @action
+  show(model) {
+    this.showing = true;
+    this.model = model;
+  }
+
+  @action
+  hide() {
+    // return tne focus back to the textarea
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/services/d-templates-modal.js:17]:
```suggestion
    // return the focus back to the textarea
```

File: assets/javascripts/discourse/services/d-templates-modal.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import { tracked } from "@glimmer/tracking";
+
+export default class DTemplatesModalService extends Service {
+  @tracked showing = false;
+  @tracked model = null;
+
+  @action
+  show(model) {
+    this.showing = true;
+    this.model = model;
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/services/d-templates-modal.js:12]: We could drop `showing`, and instead do

```
{{#if model}}
```
in the template?

**Review by davidtaylorhq — APPROVED** [review]
🚀

File: assets/javascripts/discourse/services/d-templates-modal.js (hunk 2 above)
Comment (megothss) [reply assets/javascripts/discourse/services/d-templates-modal.js]: Nice catch. Done!

**Pull request merged**